pkg/integration/utils.go

```diff
@@ -344,6 +344,21 @@ func (c *ChannelBuffer) ReadTimeout(p []byte, n time.Duration) (int, error) {
 	}
 }
 
+// FlushTimeout flush content of channel within the specified duration as timeout
+func (c *ChannelBuffer) FlushTimeout(n time.Duration) error {
+	var stop bool
+	go func() {
+		<-time.After(n)
+		stop = true
+	}()
+	for range c.C {
+		if stop {
```

Comment:
I think the problem may not exists. 
We also don't have synchronization mechanism between `Write` and `ReadTimeout`, it also works fine.
I have to say I'm a little confused about the ChannelBuffer and `C chan[]byte`, I'm not very confident about its behavior. 
I think that's also why you and I have different understanding about it.